app/src/main/java/com/simply/birthdayapp/commondata/di/CommonDataModule.kt

```diff
@@ -1,18 +1,46 @@
 package com.simply.birthdayapp.commondata.di
 
 import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.ApolloRequest
+import com.apollographql.apollo.api.ApolloResponse
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.interceptor.ApolloInterceptor
+import com.apollographql.apollo.interceptor.ApolloInterceptorChain
 import com.simply.birthdayapp.BuildConfig
 import com.simply.birthdayapp.commondata.local.DataStoreProviderImpl
 import com.simply.birthdayapp.commondomain.local.DataStoreProvider
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
 import org.koin.core.module.dsl.bind
 import org.koin.core.module.dsl.singleOf
 import org.koin.dsl.module
 
+
 val commonDataModule = module {
     singleOf(::DataStoreProviderImpl) { bind<DataStoreProvider>() }
     single<ApolloClient> {
         ApolloClient.Builder()
             .serverUrl(BuildConfig.API_URL)
+            .addInterceptor(AuthInterceptor(get()))
             .build()
     }
+}
+
+class AuthInterceptor(private val dataStoreProvider: DataStoreProvider) : ApolloInterceptor {
```

Comment:
for interceptors you should have different package for example data/network/interceptor
di modules not relates with that